src/karmen_backend/server/clients/octoprint.py

```diff
@@ -454,32 +445,36 @@ def uncached_status(self, force=True):
         else:
             return {"state": "Printer is not responding", "temperature": {}}
 
+    def _parse_device_url(self, url):
+        '''
+        normalizes device url and translates it to uri pointing websocket proxy if necessary
+        '''
+        if url is None:
+            return None
+        if url.startswith("/"):
+            return self.network_base + url
+        if not (url.startswith("http://") or url.startswith("https://")):
+            url = "http://" + url
+        parsed = urlparse.urlparse(url)
+        port = parsed.port if parsed.port is not None else "80"
+        if self.token is not None and self.token != "" and port != "80":
+            return None  # Yet we can't sed other port's than 80 over ws api yet
+        if self.token:
+            # If we have ws api, we just try the path, not much else to do
+            return "%s%s?%s" % (self.network_base, parsed.path, parsed.query)
```

Comment:
I'd prefer using of urlparse methods (like urlunparse and urlunsplit) here.

Reply:
I was not aware about urlunparse, thanks. Nevertheless, these lines were just moved, I'd keep it for now if you would'n mind.